lib/features/auth/domain/auth_bloc.dart

```diff
@@ -0,0 +1,172 @@
+    on<FacebookEvent>(_logInWithFacebook);
+    on<ShowSignUpScreenEvent>((event, emit) {
+      emit(SignUpScreenState());
+    });
+    on<ShowAuthScreenEvent>((event, emit) {
+      emit(AuthScreenState());
+    });
+
+  }
+
+  Stream<AuthState> mapEventToState(AuthEvent event) async* {
+    if (event is LoginEvent) {
+      yield AuthLoadingState();
+      try {
+        yield AuthLoadedState();
+      } catch (e) {
+        yield AuthErrorState(e.toString());
+      }
+    } else if (event is ShowSignUpScreenEvent) {
+      yield SignUpScreenState();
+    } else if (event is GoogleEvent) {
+    } else if (event is FacebookEvent) {
+    }
+  }
+
+  final FirebaseAuth auth = FirebaseAuth.instance;
+
+  Future<void> _onRegister(RegisterEvent event, Emitter<AuthState> emit) async {
+    emit(AuthLoadingState());
+    try {
+      if (event.email.isEmpty || event.password.isEmpty || event.name.isEmpty) {
+        emit(AuthErrorState('Please fill out all fields'));
+        return;
+      }
+
+      final userCredential = await FirebaseAuth.instance.createUserWithEmailAndPassword(
+        email: event.email,
+        password: event.password,
+      );
+
+      await userCredential.user?.updateProfile(displayName: event.name);
+
+      await userCredential.user?.reload();
+      final user = FirebaseAuth.instance.currentUser;
+
+      if (user?.displayName == event.name) {
+        emit(AuthLoadedState());
+      } else {
+        emit(AuthErrorState('Failed to update user profile'));
+      }
+    } on FirebaseAuthException catch (e) {
+      if (e.code == 'weak-password') {
+        emit(AuthErrorState('The password provided is too weak'));
+      } else if (e.code == 'email-already-in-use') {
+        emit(AuthErrorState('The account already exists for that email'));
+      } else {
+        emit(AuthErrorState('An unexpected error occurred.'));
+      }
+    } catch (e) {
+      emit(AuthErrorState(e.toString()));
+    }
+  }
+
+  Future<void> _onLogin(LoginEvent event, Emitter<AuthState> emit) async {
+    emit(AuthLoadingState());
+    try {
+      await auth.signInWithEmailAndPassword(
+          email: event.email, password: event.password);
+      emit(AuthLoadedState());
+    } on FirebaseAuthException catch (e) {
+      if (e.code == 'user-not-found') {
+        emit(AuthErrorState('No user found for that email.'));
+      } else if (e.code == 'wrong-password') {
+        emit(AuthErrorState('Wrong password provided for that user.'));
+      } else {
+        emit(AuthErrorState('An unexpected error occurred.'));
+      }
+    }
+  }
+
+  Future<void> _handleGoogleSignIn(
+      GoogleEvent event, Emitter<AuthState> emit) async {
+    emit(AuthLoadingState());
+    try {
+      GoogleAuthProvider googleAuthProvider = GoogleAuthProvider();
+      await auth.signInWithProvider(googleAuthProvider);
+      if (auth.currentUser != null) {
+        emit(AuthLoadedState());
+      } else {
+        (e) => emit(AuthErrorState(e.toString()));
```

Comment:
Unnecessary anonim function . argument always will be null

Reply:
Done